app/src/test/java/org/simple/clinic/summary/bloodsugar/BloodSugarSummaryViewEffectHandlerTest.kt

```diff
@@ -1,20 +1,30 @@
 package org.simple.clinic.summary.bloodsugar
 
+import com.nhaarman.mockito_kotlin.any
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.whenever
+import io.reactivex.Observable
 import org.junit.Test
+import org.simple.clinic.bloodsugar.BloodSugarMeasurement
+import org.simple.clinic.bloodsugar.BloodSugarRepository
 import org.simple.clinic.mobius.EffectHandlerTestCase
+import org.simple.clinic.util.scheduler.TrampolineSchedulersProvider
 import java.util.UUID
 
 class BloodSugarSummaryViewEffectHandlerTest {
 
   @Test
   fun `when fetch blood sugar effect is received then blood sugar should be fetched`() {
     //given
-    val testCase = EffectHandlerTestCase(BloodSugarSummaryViewEffectHandler.create())
+    val bloodSugarRepository = mock<BloodSugarRepository>()
+    val testCase = EffectHandlerTestCase(BloodSugarSummaryViewEffectHandler.create(bloodSugarRepository, TrampolineSchedulersProvider()))
+    val measurements = listOf<BloodSugarMeasurement>()
+    whenever(bloodSugarRepository.latestMeasurements(any(), any())).thenReturn(Observable.just(measurements))
```

Comment:
Instead of using catch-all matchers (`any()`), we should be using the values that we expect to be called. Otherwise, this might end up passing when it should fail.

An example would be if the limit to read blood sugars was being passed via a config, but the production source was not actually using the config and just hardcoding the value. Letting it be `any()` would make the test pass regardless of whether the parameter was read from the config or not and will hide a bug.

We could make this `whenever(bloodSugarRepository.latestMeasurements(patientUuid, 100)).thenReturn(Observable.just(measurements))` and then change it later to the config value when we implement the config.